### ISSUE TITLE
Mypy annotation for dictionary needs explicit import

### ISSUE THREAD

**Issue by ArneBachmann** [evhub/coconut#339]
```
try: import typing
except: pass  # Python 2
def func(a:typing.Dict[int, int]) -> None
  pass
```
because Dict is not auto-imported and typing doesn't exist in Python 2

**Comment by evhub** [comment]
@ArneBachmann What sort of syntax would you like to see instead? Just `typing.Dict` without requiring the import sems too implicit to me.

**Comment by ArneBachmann** [comment]
Let's say I'm just surprised that type annotations which are checked from an external tool require imports in my actual runtime code. Maybe it's simply the way Python or mypy work, but it seemed counter-intuitive, especially since type indicators `str`, `int` are always imported, while `Dict`, `Tuple`, `Union` are handled in a different way.

**Comment by evhub** [comment]
@ArneBachmann Solved by #383.

**Comment by evhub** [comment]
Replaced by #403.
